### PR TITLE
Run pytest on macos in GH actions

### PR DESCRIPTION
**Pull request by reuterbal**
With people using Loki locally on their MacBooks, it seems useful to test this as part of the CI.

### CODE REVIEW

**Comment by reuterbal** [comment]
I think this is ready for a look. I had to do a few things in addition to ensure compatibility with MacOS, but I think it generally helps to increase being platform-agnostic.

In detail:

- Improved the CMake recipes, using CMake-internal utilities for the Python environment hacking instead of relying on GNU sed syntax
- CMake install now ensures it picks a Python version that is supported
- Added installation instructions for MacOS and updated them for Linux, too
- I had to add a bit more tolerance in the timeout test
- I had to disable test_transform_normalize_array_shape_and_access on MacOS. This looks like it might be a genuine memory corruption problem that only triggers on MacOS - documented in #352 for further investigation (but orthogonal to this PR, I think).

**Review by mlange05 — APPROVED** [review]
Looks good to me, and very useful in general. Thanks for grinding this one out! :pray: 

Only question I have a is about the tuple-comparison changes, but otherwise this looks  great. GTG from me once that is answered.

File: loki/batch/tests/test_batch.py [hunk 1]
```diff
@@ -249,7 +249,7 @@ def test_file_item2(testdir):
     )
 
     # Files don't have dependencies (direct dependencies, anyway)
-    assert item.dependencies is ()
+    assert isinstance(item.dependencies, tuple) and not item.dependencies
```
[the remark this answers is not in the document]
Comment (reuterbal) [reply loki/batch/tests/test_batch.py:252]: Pytest doesn't like `is` comparison against literals:
```
 /home/runner/work/loki/loki/loki/batch/tests/test_batch.py:252: SyntaxWarning: "is" with a literal. Did you mean "=="?
  assert item.dependencies is ()
```
See for example here: https://github.com/ecmwf-ifs/loki/actions/runs/10124810197/job/27999587354#step:8:120

[the remark this answers is not in the document]
Comment (mlange05) [reply loki/batch/tests/test_batch.py:252]: Ok, good to know. Thanks.

**Pull request merged**